### PR TITLE
feat(individual)!: check that page is publishable (Resolves #594)

### LINKED ISSUE #594: Individual: check that page is publishable

### PR DESCRIPTION
**Pull request by jobara**
Ensures that required fields in an Individual's model have been filled in before allowing allowing the isPublishable method to return true. Also uses this method to only show the "Publish" button for individual pages after the criteria has been met.

Resolves #594 

### Prerequisites

If this PR changes PHP code or dependencies:

- [x] I've run `composer format` and fixed any code formatting issues.
- [x] I've run `composer analyze` and addressed any static analysis issues.
- [x] I've run `php artisan test` and ensured that all tests pass.
- [x] I've run `composer localize` to update localization source files and committed any changes.

If this PR changes CSS or JavaScript code or dependencies:

- [x] I've run `npm run lint` and fixed any linting issues.
- [ ] I've run `npm run build` and ensured that CSS and JavaScript assets can be compiled.

### CODE REVIEW

**Comment by jobara** [comment]
@greatislander I did add a test for what I believed should test the policy change, but the test coverage was still reporting a line being missed. Perhaps something else is blocking it as well? Any thoughts?

File: tests/Feature/IndividualTest.php [hunk 1]
```diff
@@ -756,6 +774,52 @@
...
+test('individual pages cannot be published by other users', function () {
+    $user = User::factory()->create();
+    $individual = Individual::factory()->create([
+        'roles' => ['consultant'],
+        'published_at' => null,
+    ]);
+
+    $response = $this->actingAs($user)->from(localized_route('individuals.show', $individual))->put(localized_route('individuals.update-publication-status', $individual), [
+        'publish' => true,
+    ]);
```
Comment (jobara) [inline tests/Feature/IndividualTest.php]: @greatislander I tried removing the `from` portion, as per our conversation but it still wasn't hitting the publish policy. I believe the `put` will actually fail on the `update` policy from the 'individuals.update-publication-status' route. So I'm not sure it's actually possible to hit the line of code. I'm going to collapse the policy back down and for now revert back to the old message as it's more generic in case something does slip through.

File: app/Policies/IndividualPolicy.php [hunk 2]
```diff
@@ -63,9 +63,13 @@ public function update(User $user, Individual $individual): Response
...
     {
-        return $user->id === $individual->user_id && $individual->isPublishable()
+        if ($user->id !== $individual->user_id) {
+            return Response::deny(__('You cannot publish this individual page.'));
+        }
+
+        return $individual->isPublishable()
             ? Response::allow()
-            : Response::deny(__('You cannot publish this individual page.'));
+            : Response::deny(__('This individual page is missing some required information, and cannot be published yet.'));
```
Comment (jobara) [inline app/Policies/IndividualPolicy.php]: It doesn't appear that it is actually possible to hit the first condition. I've tried to write tests for it, but it gets caught on other policies first. I'm planning to revert this change to keep the more generic message just in case something does slip through.

File: resources/views/individuals/partials/progress.blade.php [hunk 3]
```diff
@@ -23,7 +23,8 @@
         @if ($individual->checkStatus('draft'))
             <p class="stack">
                 <button class="secondary" name="preview" value="1">{{ __('Preview page') }}</button>
-                <button class="secondary" name="publish" value="1">{{ __('Publish page') }}</button>
+                <button class="secondary" name="publish" value="1"
+                    @cannot('publish', $individual) disabled @endcannot>{{ __('Publish page') }}</button>
```
Comment (greatislander) [inline resources/views/individuals/partials/progress.blade.php:27]: I was thinking following our earlier conversation, we'll need to address this but probably should do so in a separate PR to keep this one focused.

File: tests/Datasets/IndividualIsPublishable.php [hunk 4]
```diff
@@ -0,0 +1,393 @@
+<?php
+
+use App\Enums\CommunityConnectorHasLivedExperience;
+use App\Enums\ConsultingService;
+use App\Enums\IndividualRole;
+use App\Enums\MeetingType;
+
+dataset('individualIsPublishable', function () {
```
Comment (greatislander) [inline tests/Datasets/IndividualIsPublishable.php:8]: Nice use of datasets! This is new to me but perfect for this situation.

**Review by greatislander — APPROVED** [review]
This looks great— thanks, @jobara!
